nodewatcher/modules/analysis/rogue_nodes/algorithm.py

```diff
@@ -0,0 +1,39 @@
+import networkx as nx
+
+
+def rogue_node_detection_algorithm(graph, friendly_nodes):
+    """
+    Detects rogue nodes and outputs an array of detected rogue nodes in the graph. The algorithm first creates a
+    minimum spanning tree of the graph and then computes the probability of a node being rogue from
+     the number of edges that node has in a MST. The algorithm takes at most O(E log(V)) time.
+
+    :param graph: A graph datastructure which contains both friendly and unknown nodes.
+    :param friendly_nodes: An array of nodes that are monitored by nodewatcher and are considered friendly.
+    :return: An array of all unknown nodes with an associated probability of being a rogue node.
+    """
+
+    nx_graph = nx.Graph()
+    nx_graph.add_nodes_from([(node['i'], {'b': node['b']}) if 'b' in node else node['i'] for node in graph['v']])
```

Comment:
Yes, that's how you create nodes in nx:
http://networkx.readthedocs.io/en/latest/reference/generated/networkx.Graph.add_node.html#networkx.Graph.add_node

It has to be a tuple in case you're storing additional information.